### PR TITLE
Add notebook for setting up LLM training environment

### PR DESCRIPTION
**Pull request by meoflynn**
This PR adds a new LLM specific notebook. This is similar to the pytorch notebook, but with a script containing installation steps that are required for the training environment.

### CODE REVIEW

**Review by anish-mudaraddi — APPROVED** [review]

File: jupyter-llm-notebook/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY install-sciml-deps.sh /home/${NB_USER}
+RUN chown ${NB_UID} /home/${NB_USER}/install-sciml-deps.sh
+RUN chmod +rx /home/${NB_USER}/install-sciml-deps.sh
+
+USER $NB_UID
```
Comment (Chris-green-stfc) [inline jupyter-llm-notebook/Dockerfile:15]: Why is this line `$NB_UID` while other lines use `${NB_UID}`? Will that make a difference?

Comment (meoflynn) [reply jupyter-llm-notebook/Dockerfile:15]: There is not much difference between them, it's the pattern that's in the other notebooks as well so I kept it consistent here

Comment (anish-mudaraddi) [reply jupyter-llm-notebook/Dockerfile:15]: While technically you can do that - for simplicity you don't need to - makes it easier to read

Comment (meoflynn) [reply jupyter-llm-notebook/Dockerfile:15]: The curly braces protect the variable, it's optional to include curly braces around the variable.

**Pull request closed without merging**